modules/grandpa/src/lib.rs

```diff
@@ -141,8 +119,8 @@ pub mod pallet {
 		/// If successful in verification, it will write the target header to the underlying storage
 		/// pallet.
 		#[pallet::weight(T::WeightInfo::submit_finality_proof(
-			T::MaxBridgedSessionLength::get().as_() as u32,
-			T::MaxBridgedValidatorCount::get(),
+			justification.commit.precommits.len() as u32,
```

Comment:
Is this correct order of arguments? From the trait it seems that the first argument is `v` (`let v in 1..MAX_VOTE_ANCESTRIES;`).

Reply:
Oh wow, you're right these are flipped. Good catch!